core/serializers.py

```diff
@@ -49,13 +49,13 @@ def validate(self, attrs):
         except Hackathon.DoesNotExist:
             raise exceptions.NotFound(detail="Hackathon does not exist!")
         try:
-            team = Team.objects.get(hackathon=hackathon, members=user)
+            team = Team.objects.select_related('leader', 'members', 'hackathon').get(hackathon=hackathon, members=user)
```

Comment:
I think you weren't able to actually get the gist of why we use `select_related` its for use cases when we will later on try to access a field of the foreign attribute whereas in the validation function we aren't accessing any field of `leader`, `member` or `hackathon`. So there is no need to increase overhead with `select_related`